src/test/javascript/spec/component/exercises/shared/submission-export-dialog.component.spec.ts

```diff
@@ -0,0 +1,142 @@
+            providers: [
+                { provide: SubmissionExportService, useValue: MockProvider(SubmissionExportService) },
+                { provide: ExerciseService, useClass: MockExerciseService },
+            ],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(SubmissionExportDialogComponent);
+                component = fixture.componentInstance;
+
+                alertService = TestBed.inject(AlertService);
+                submissionExportService = TestBed.inject(SubmissionExportService);
+                exerciseService = TestBed.inject(ExerciseService);
+
+                component.exerciseId = exerciseId;
+                component.exerciseType = validExerciseType;
+                component.submissionExportOptions = submissionExportOptions;
+            });
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+
+    it('should initialize', () => {
+        const findSpy = jest.spyOn(exerciseService, 'find');
+        component.ngOnInit();
+
+        expect(component.isLoading).toBe(false);
+        expect(component.exportInProgress).toBe(false);
+        expect(component.submissionExportOptions).toEqual(submissionExportOptions);
+        expect(component.exercise).toEqual({ id: exerciseId } as Exercise);
+        expect(findSpy).toHaveBeenCalledTimes(1);
+        expect(findSpy).toHaveBeenCalledWith(exerciseId);
+    });
+
+    it('should handle export response', () => {
+        const response: HttpResponse<Blob> = new HttpResponse();
+        const alertSpy = jest.spyOn(alertService, 'success');
+        const modalSpy = jest.spyOn(component.activeModal, 'dismiss');
+        const downloadSpy = jest.spyOn(DownloadUtil, 'downloadZipFileFromResponse');
+
+        component.handleExportResponse(response);
+
+        expect(alertSpy).toHaveBeenCalledTimes(1);
+        expect(alertSpy).toHaveBeenCalledWith('instructorDashboard.exportSubmissions.successMessage');
+        expect(modalSpy).toHaveBeenCalledTimes(1);
+        expect(modalSpy).toHaveBeenCalledWith(true);
+        expect(component.exportInProgress).toBe(false);
+        expect(downloadSpy).toHaveBeenCalledTimes(1);
+        expect(downloadSpy).toHaveBeenCalledWith(response);
+    });
+
+    it('should clear dialog', () => {
+        const modalSpy = jest.spyOn(component.activeModal, 'dismiss');
+        component.clear();
+
+        expect(modalSpy).toHaveBeenCalledTimes(1);
+        expect(modalSpy).toHaveBeenCalledWith('cancel');
+    });
+
+    describe('Exporting Submission', () => {
+        let exportSubmissionServiceMock: jest.Mock;
+        let handleExportResponseMock: jest.Mock;
+
+        beforeEach(() => {
+            handleExportResponseMock = jest.fn().mockReturnValue(of());
+            component.handleExportResponse = handleExportResponseMock;
+        });
+
+        afterEach(() => {
+            jest.restoreAllMocks();
+        });
```

Comment:
```suggestion
```
This isn't necessary, as you're already doing this in the 'beforeEach' block of the outer 'describe' block

Reply:
Implemented.